### PR TITLE
Validate Node model fields

### PR DESCRIPTION
**Pull request by andbortnik**

### CODE REVIEW

File: thenewboston_node/business_logic/validators.py [hunk 1]
```diff
@@ -86,3 +93,30 @@ def validate_exact_value(subject, value, expected_value):
...
+    except ValueError:
+        return False
+
+
+def is_fqdn(value):
+    return fqdn.FQDN(value).is_valid
+
+
+def is_uri(value):
+    uri = uritools.urisplit(value)
```
Comment (dmugtasimov) [inline thenewboston_node/business_logic/validators.py:121]: why not https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse ?

Comment (andbortnik) [reply thenewboston_node/business_logic/validators.py]: `uritools` docs states that `urlparse` is not fully RFC 3986 compliant due to backward compatibility but I didn't find which part of it. Changed it to `urlparse` for now.

File: thenewboston_node/business_logic/models/signed_change_request_message/node_declaration.py [hunk 2]
```diff
@@ -27,3 +28,7 @@ def create(
                 fee_account=fee_account,
             )
         )
+
+    @validates('node declaration signed change request message')
```
Comment (dmugtasimov) [inline thenewboston_node/business_logic/models/signed_change_request_message/node_declaration.py:32]: should not we have `validate_node()` and skip explicit text?

File: thenewboston_node/business_logic/models/node.py [hunk 3]
```diff
@@ -23,6 +27,35 @@ class Node(BaseDataclass):
...
     )
 
+    @validates('node')
+    def validate(self):
+        self.validate_identifier()
+        self.validate_fee_amount()
+        self.validate_fee_account()
+        self.validate_network_addresses()
+
+    @validates('node identifier')
```
Comment (dmugtasimov) [inline thenewboston_node/business_logic/models/node.py:37]: Do we really need explicit text?

File: thenewboston_node/business_logic/models/node.py [hunk 4]
```diff
@@ -23,6 +27,35 @@ class Node(BaseDataclass):
...
+    def validate_identifier(self):
+        validate_not_empty(f'{self.humanized_class_name} identifier', self.identifier)
+        validate_type(f'{self.humanized_class_name} identifier', self.identifier, str)
+
+    @validates('node network addresses')
```
Comment (dmugtasimov) [inline thenewboston_node/business_logic/models/node.py:42]: Do we really need explicit text?

File: thenewboston_node/business_logic/models/node.py [hunk 5]
```diff
@@ -23,6 +27,35 @@ class Node(BaseDataclass):
...
+    def validate_network_addresses(self):
+        for network_address in self.network_addresses:
+            validate_not_empty(f'{self.humanized_class_name} network_addresses', network_address)
+            validate_network_address(f'{self.humanized_class_name} network_addresses', network_address)
+
+    @validates('node fee amount')
```
Comment (dmugtasimov) [inline thenewboston_node/business_logic/models/node.py:48]: Do we really need explicit text?

File: thenewboston_node/business_logic/models/node.py [hunk 6]
```diff
@@ -23,6 +27,35 @@ class Node(BaseDataclass):
...
+    def validate_fee_amount(self):
+        validate_not_none(f'{self.humanized_class_name} fee_amount', self.fee_amount)
+        validate_type(f'{self.humanized_class_name} fee_amount', self.fee_amount, int)
+        validate_gte_value(f'{self.humanized_class_name} fee_amount', self.fee_amount, 0)
+
+    @validates('node fee account')
```
Comment (dmugtasimov) [inline thenewboston_node/business_logic/models/node.py:54]: Do we really need explicit text?

File: thenewboston_node/business_logic/models/node.py [hunk 7]
```diff
@@ -23,6 +27,35 @@ class Node(BaseDataclass):
         default=None, metadata={'example_value': '7a5dc06babda703a7d2d8ea18d3309a0c5e6830a25bac03e69633d283244e001'}
     )
 
+    @validates('node')
```
Comment (dmugtasimov) [inline thenewboston_node/business_logic/models/node.py:30]: Do we really need explicit text?

File: thenewboston_node/business_logic/models/node.py [hunk 8]
```diff
@@ -23,6 +27,35 @@ class Node(BaseDataclass):
...
+    def validate(self):
+        self.validate_identifier()
+        self.validate_fee_amount()
+        self.validate_fee_account()
+        self.validate_network_addresses()
+
+    @validates('node identifier')
+    def validate_identifier(self):
+        validate_not_empty(f'{self.humanized_class_name} identifier', self.identifier)
+        validate_type(f'{self.humanized_class_name} identifier', self.identifier, str)
```
Comment (dmugtasimov) [inline thenewboston_node/business_logic/models/node.py:40]: I think that we should also validate that is contains only hexadecimal characters

Comment (dmugtasimov) [reply thenewboston_node/business_logic/models/node.py:40]: same for fee_account

File: thenewboston_node/business_logic/tests/test_models/test_signed_change_request/test_node_declaration_signed_change_request.py [hunk 9]
```diff
@@ -14,4 +18,134 @@ def test_can_create_node_declaration_signed_change_request(user_account_key_pair
...
+
+@pytest.mark.parametrize(
+    'network_address', [
+        'http://127.0.0.1',
+        'http://127.0.0.1:8080',
+        'http://[2001:db8::123.123.123.123]:80',
+        'http://xn--d1acufc.xn--p1ai',
+        'http://example.com',
+        'https://my.domain.com',
+        'https://my.domain.com/path/to/resource#fragment',
```
Comment (dmugtasimov) [inline thenewboston_node/business_logic/tests/test_models/test_signed_change_request/test_node_declaration_signed_change_request.py:41]: add `'http://localhost:8555/'`

**Pull request merged**